### ISSUE TITLE
Outline and Border-radius are ignored on Safari

### ISSUE THREAD

**Issue by hiddevdploeg** [GoogleWebComponents/model-viewer#662]
## Description
When you want to change `outline` and `border-radius` in css to the `model-viewer` object, it is being ignored in Safari while it works on Chrome.

```
    model-viewer {
      border-radius: 1rem;
      outline: none;
    }
```

### Browser Affected

- [ ] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] Helios
- [ ] IE
- [x] Safari

### OS

- [ ] Android
- [ ] iOS
- [ ] Linux
- [x] MacOS
- [ ] Windows

### Versions

- model-viewer: vX.X.X
- three.js: rXX

**Comment by cdata** [comment]
Hey, thanks for filing this.

## `border-radius`

**TL;DR** try setting `transform: translateZ(0)` (or similar) on the element that you set `border-radius` on and see if it fixes the issue!

The `border-radius` thing is quite unfortunate. I tried to track down potential Webkit bugs that could be related. The most relevant thing I could find was this really old Webkit bug (https://bugs.webkit.org/show_bug.cgi?id=115613), which seems relevant but has ambiguous implications at best.

I played around with setting `border-radius` in various ways to see what works. Apparently it's possible to set the `border-radius` properly if we set it directly on the `<canvas>`. That was the first clue.

The second clue came when I crossed [this Stack Overflow](https://stackoverflow.com/questions/10616668/how-to-hide-canvas-content-from-parent-rounded-corners-in-any-webkit-for-mac/12336233#12336233) from forever ago. The "fix" in that article has bad side-effects, but when I tried it out, that's when it clicked that this is probably a render layer issue.

Sure enough, changing the clipping ancestor (that is, the thing you set `border-radius` on) into a render layer fixed the `border-radius` problem.

## `outline`

On the topic of outline, we just landed #660 . It's not released yet, but if you try using [`<model-viewer>` built from master](https://googlewebcomponents.github.io/model-viewer/dist/model-viewer.js) and also put [the `:focus-visible` polyfill](https://unpkg.com/focus-visible@5.0.1/dist/focus-visible.js) on your page, the `outline` will only appear in cases where `<model-viewer>` was focused due to keyboard interaction (which is when you usually want it to appear for a11y purposes).

The fundamental issue is that the element that needs to be styled is different in Safari (it's the `<canvas>` in the shadow root). This is possibly attributable to [Safari's missing support for `delegatesFocus`](https://bugs.webkit.org/show_bug.cgi?id=166484).

In the near term, one thing we could do to help folks is to add a CSS custom property for styling the canvas outline. An alternative would be to come up with a combination of styles that allow us to always disable the `outline` on the `<canvas>`. I'm open to suggestions for good ways to approach this problem.

**Comment by cdata** [comment]
@hiddevdploeg I'm gonna go ahead and close this, as #660 has landed and it seems like there is a path forward for the points you raised. Please ping the issue again if you would like it re-opened!

**Comment by nolimitdev** [comment]
@Ramzes-org transform: translateZ(0) does not fixes this Safari problem in my case. Are you sure? Create JS fiddle demo please. Thanks.
